packages/typespec-client-generator-core/test/public-utils.test.ts

```diff
@@ -1404,13 +1426,13 @@ describe("typespec-client-generator-core: public-utils", () => {
         deepStrictEqual(
           models.map((x) => x.name).sort(),
           [
-            "ResourceOperationStatus",
+            "ResourceOperationStatusUserExportedUserError",
             "OperationState",
             "Error",
             "InnerError",
             "ExportedUser",
             "ErrorResponse",
-            "OperationStatus",
+            "OperationStatusExportedUserError",
```

Comment:
i believe these are both core models, correct?
nit: I don't think we should modify the names of core models based off of templating. Not the biggest issue though, bc ideally no one should be generating with core models. If it makes the code much harder, no need to change your existing code

Reply:
AFAIK, `OperationStatus` is exported in Java and .NET. I don't think this will only happen for core model. This will happen when implicitly using templated model without `@friendlyName` in templated operation. And it is a possible usage for customer. Since .NET and Java need this nonredundant naming insurance, I prefer to add it in TCGC.